server/graphql/common/expenses.ts

```diff
@@ -1643,21 +1645,28 @@ export const checkHasBalanceToPayExpense = async (
   { forceManual = false, manualFees = {}, useExistingWiseData = false } = {},
 ) => {
   const payoutMethodType = payoutMethod ? payoutMethod.type : expense.getPayoutMethodTypeFromLegacy();
-  const balanceInExpenseCurrency = await expense.collective.getBalanceWithBlockedFunds({ currency: expense.currency });
+  const balanceInCollectiveCurrency = await expense.collective.getBalanceWithBlockedFunds();
   const isSameCurrency = expense.currency === expense.collective.currency;
+  const exchangeStats =
+    !isSameCurrency && (await models.CurrencyExchangeRate.getPairStats(expense.collective.currency, expense.currency));
 
-  // Ensure the collective has enough funds to pay the expense, with an error margin of 20% of the expense amount
-  // to account for fluctuating rates. Example: to pay for a $100 expense in euros, the collective needs to have at least $120.
-  const getMinExpectedBalance = amountToPay => (isSameCurrency ? amountToPay : Math.round(amountToPay * 1.2));
+  // Ensure the collective has enough funds to pay the expense, with an error margin of 2σ (standard deviations) the exchange rate of past 5 days
+  // to account for fluctuating rates. (2σ counts for 95% of rates of the given period)
+  const getMinExpectedBalance = amountToPayInExpenseCurrency =>
+    isSameCurrency
+      ? amountToPayInExpenseCurrency
+      : Math.round(amountToPayInExpenseCurrency / (exchangeStats.latestRate - exchangeStats.stddev * 2));
```

Comment:
`getMinExpectedBalance` can return null if we don't have any cached information between from/to currencies. I would move all this to a separate function to isolate complexity, with an eventual fallback to the legacy rule of 20% if we're not able to compute anything better,

Reply:
@Betree, I'm not abstracting this to a global scope because I don't think this will be very useful elsewhere.
I converted this to an assertion function and included the error message format in it to simplify it further.